python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py

```diff
@@ -0,0 +1,211 @@
+from pprint import pformat
+from typing import Any, Dict, List, Optional, cast
+
+import boto3
+import botocore
+import dagster._check as check
+from dagster import PipesClient
+from dagster._annotations import experimental
+from dagster._core.definitions.resource_annotation import TreatAsResourceParam
+from dagster._core.errors import DagsterExecutionInterruptedError
+from dagster._core.execution.context.compute import OpExecutionContext
+from dagster._core.pipes.client import (
+    PipesClientCompletedInvocation,
+    PipesContextInjector,
+    PipesMessageReader,
+)
+from dagster._core.pipes.utils import PipesEnvContextInjector, open_pipes_session
+from mypy_boto3_ecs.client import ECSClient
+from mypy_boto3_ecs.type_defs import RunTaskRequestRequestTypeDef
+from typing_extensions import Unpack
+
+from dagster_aws.pipes.message_readers import PipesCloudWatchMessageReader
+
+
+@experimental
+class PipesECSClient(PipesClient, TreatAsResourceParam):
+    """A pipes client for running AWS ECS tasks.
+
+    Args:
+        client (Optional[boto3.client]): The boto ECS client used to launch the ECS task
+        context_injector (Optional[PipesContextInjector]): A context injector to use to inject
+            context into the ECS task. Defaults to :py:class:`PipesEnvContextInjector`.
+        message_reader (Optional[PipesMessageReader]): A message reader to use to read messages
+            from the ECS task. Defaults to :py:class:`PipesCloudWatchMessageReader`.
+        forward_termination (bool): Whether to cancel the ECS task when the Dagster process receives a termination signal.
+    """
+
+    def __init__(
+        self,
+        client: Optional[ECSClient] = None,
+        context_injector: Optional[PipesContextInjector] = None,
+        message_reader: Optional[PipesMessageReader] = None,
+        forward_termination: bool = True,
+    ):
+        self._client = client or boto3.client("ecs")
+        self._context_injector = context_injector or PipesEnvContextInjector()
+        self._message_reader = message_reader or PipesCloudWatchMessageReader()
+        self.forward_termination = check.bool_param(forward_termination, "forward_termination")
+
+    @classmethod
+    def _is_dagster_maintained(cls) -> bool:
+        return True
+
+    def run(
+        self,
+        *,
+        context: OpExecutionContext,
+        task_definition: str,
+        extras: Optional[Dict[str, Any]] = None,
+        **params: Unpack[RunTaskRequestRequestTypeDef],
+    ) -> PipesClientCompletedInvocation:
+        """Start a ECS task, enriched with the pipes protocol.
+
+        See also: `AWS API Documentation <https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RunTask.html>`_
+
+        Returns:
+            PipesClientCompletedInvocation: Wrapper containing results reported by the external
+            process.
```

Comment:
Write docs, especially for the params part.

Reply:
There is another PR for docs: https://github.com/dagster-io/dagster/pull/23569

Reply:
I just included docstrings in the current PR for convenience